trace/client_test.go

```diff
@@ -139,6 +140,55 @@ func TestUNIXBuffered(t *testing.T) {
 	}
 }
 
+func TestUNIXBufferedFlushing(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test_unix")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	sockName := filepath.Join(dir, "sock")
+	laddr, err := net.ResolveUnixAddr("unix", sockName)
+	require.NoError(t, err)
+
+	outPkg := make(chan *protocol.Message, 4)
+	cleanup := serveUNIX(t, laddr, func(in net.Conn) {
+		for {
+			pkg, err := protocol.ReadSSF(in)
+			if err == io.EOF {
+				return
+			}
+			assert.NoError(t, err)
+			outPkg <- pkg
+		}
+	})
+	defer cleanup()
+	flushChan := make(chan time.Time)
+
+	client, err := NewClient((&url.URL{Scheme: "unix", Path: sockName}).String(),
+		Capacity(4),
+		Buffered,
+		FlushChannel(flushChan, func() {}))
+	require.NoError(t, err)
+	defer client.Close()
```

Comment:
Should `Close` also flush the buffer? Otherwise a client may exit with things in the buffer if the ticker hasn't fired?

Reply:
Great question! `Close` is a synchronous operation (it comes in on the same pipe as SSF writes, flushes, etc.). So yup, if we get to the point where the context gets canceled, we're assuming to not have a working endpoint anymore.

AFAICT, it's true that `Close` does lose the data written to the buffer before it could be flushed; I figured it'd be more important to close the client in a timely fashion than make sure it's completely flushed.